app/controllers/concerns/legacy/workflow_support.rb

```diff
@@ -11,6 +11,7 @@ module WorkflowSupport
     def create_ro_crate
       @crate_builder = Legacy::WorkflowCrateBuilder.new(legacy_ro_crate_params)
       @workflow.workflow_class = @crate_builder.workflow_class = WorkflowClass.find_by_id(params[:workflow_class_id])
+      @workflow.project_ids |= params[:workflow] ? params[:workflow][:project_ids] : params[:project_ids] || []
```

Comment:
I don't think this is in the right place, it doesn't belong as part of `create_ro_crate `and should have been set already.
I think you just need to add `workflow[project_ids]` to the permitted params in legacy_ro_crate_params and also possibly update `legacy_set_workflow`

for what reason do we need to handle `params[:project_ids]`, rather than it always being `workflow[project_ids]` ?

Reply:
Agree with @stuzart here. Also for the actions in WorkflowsController.

It's possible to do it all via the params, but unfortunately you have to do something slightly different in each case.